### ISSUE TITLE
Implicitly marking parameter $context as nullable is deprecated, the explicit nullable type must be used instead

### ISSUE THREAD

**Issue by BraisedFish** [barryvdh/laravel-ide-helper#1632]
### Versions:
- ide-helper Version: 3.3.0
- Laravel Version: 11.36.1 
- PHP Version: 8.4.1

### Description:

Some warnings about deprecation still exist. Run ide-helper will report this to storage\logs\php-deprecation-warnings.log.

[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock::__construct(): Implicitly marking parameter $context as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock.php on line 72  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock::__construct(): Implicitly marking parameter $location as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock.php on line 72  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Description::__construct(): Implicitly marking parameter $docblock as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Description.php on line 41  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Description::setDocBlock(): Implicitly marking parameter $docblock as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Description.php on line 193  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::createInstance(): Implicitly marking parameter $docblock as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 114  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::createInstance(): Implicitly marking parameter $location as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 114  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::__construct(): Implicitly marking parameter $docblock as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 194  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::__construct(): Implicitly marking parameter $location as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 194  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::setDocBlock(): Implicitly marking parameter $docblock as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 326  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Tag::setLocation(): Implicitly marking parameter $location as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Tag.php on line 350  
[2024-12-19 16:56:59] local.WARNING: Barryvdh\Reflection\DocBlock\Type\Collection::__construct(): Implicitly marking parameter $context as nullable is deprecated, the explicit nullable type must be used instead in D:\var\gitlab\blocgo\backend-v2\vendor\barryvdh\reflection-docblock\src\Barryvdh\Reflection\DocBlock\Type\Collection.php on line 61  

### Steps To Reproduce:

- …

**Comment by HarbourDesign** [comment]
You'll see this with 8.4 but not <= 8.3.14 versions: [.](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated).  I'm seeing this 'notice' with pusher/pusher-php-server , yet the program works fine for me.

**Comment by BraisedFish** [comment]
> You'll see this with 8.4 but not <= 8.3.14 versions: [php website](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated). I'm seeing this 'notice' with pusher/pusher-php-server , yet the program works fine for me.

Yes, it's just a warning level. The code still could run. It's just because I see the same problem was fixed in 3.3.0 version. Mabey 
the next version will do something for others.

**Comment by PNardman** [comment]
Regardless of being just warning numbers it can lead to fludding logs and issues in php 8.4 compatibility. I would strongly vote for improving this. 👍
